lte/gateway/c/core/oai/tasks/sgw/sgw_context_manager.cpp

```diff
@@ -196,36 +177,42 @@ sgw_cm_create_bearer_context_information_in_collection(teid_t teid) {
 }
 
 //-----------------------------------------------------------------------------
-hashtable_rc_t sgw_cm_remove_bearer_context_information(teid_t teid,
-                                                        imsi64_t imsi64) {
-  hashtable_rc_t temp = HASH_TABLE_OK;
-
-  hash_table_ts_t* state_teid_ht = get_spgw_teid_state();
-  temp = hashtable_ts_free(state_teid_ht, teid);
-  if (temp != HASH_TABLE_OK) {
+magma::proto_map_rc_t sgw_cm_remove_bearer_context_information(
+    teid_t teid, imsi64_t imsi64) {
+  magma::proto_map_rc_t temp = magma::PROTO_MAP_OK;
+
+  state_teid_map_t* state_teid_map = get_spgw_teid_state();
+  if (!state_teid_map) {
+    OAILOG_ERROR(LOG_SPGW_APP, "Failed to get state_teid_map");
+    return magma::PROTO_MAP_KEY_NOT_EXISTS;
+  }
+  if (state_teid_map->remove(teid) != magma::PROTO_MAP_OK) {
     OAILOG_ERROR_UE(LOG_SPGW_APP, imsi64,
-                    "Failed to free teid from state_teid_ht \n");
-    return temp;
+                    "Failed to free teid from state_teid_map \n");
+    return magma::PROTO_MAP_KEY_ALREADY_EXISTS;
```

Comment:
This should be an explicit error, not that key already exists. Can we add an error code for `PROTO_MAP_REMOVE_KEY_FAILED`

Reply:
The above comment is addressed at https://github.com/magma/magma/pull/13895/files#diff-8dbb5820f4aef5a4b6295c1040d13d405728e409346d63b37cfac400b8004312R189